### ISSUE TITLE
[2.0.x] M600 inside gcode doesn't stop in correct position

### ISSUE THREAD

**Issue by GMagician** [MarlinFirmware/Marlin#11431]
I use cura as slicer and in gcode I have layer references. At beginning of layer 172 I wrote M600 (change filament).
Printer stopped at layer 174...is it normal?

EDIT: I'm sure about not stopping in correct position because piece is 0,2mm x layer, last 1mm is a closed part and I have to insert a screw into the piece. I see printer starting to close hole so it goes over layer 172.

**Comment by ManuelMcLure** [comment]
Try inserting an `M400` gcode before the `M600` so the planer syncs up with the actual motion.

**Comment by GMagician** [comment]
M400 just calls `planner.synchronize()` and `pause_print` also calls it (it's called by M600) so I think it will not solve, but I'll try

**Comment by thinkyhead** [comment]
It should execute `M600` at the point it appears in the G-code. How are you reckoning the layers? Note that `M600` will raise Z by a couple of millimeters as part of pausing the print.

**Comment by GMagician** [comment]
I'm almost sure about layer position, I've done a double check:
1) Cura says 176 total layers. I print with 0,2mm layer height. Piece will close screw hole 1mm before end of piece. This will give me 5 layers to close it. 176-5 = 171 then 172 is where it start to close.
2) Cure let you see layer by layer what happens. at layer 172 I see it close the hole.

I had to pause manually by LCD when I saw it was closing hole without stopping. Then I inserted screw and restarted print (really I need only a pause but even M125 doesn't work). After restaring pring. Printer stopped some layer after 172 (I saw 174 on repetier...yes I use cura for slice and repetier as host :-D)
